src/Microsoft.ML.Data/Transforms/ValueToKeyMappingTransformer.cs

```diff
@@ -789,38 +789,86 @@ private bool SaveAsOnnxCore(OnnxContext ctx, int iinfo, ColInfo info, string src
                 OnnxNode castNode;
                 var labelEncoderOutput = ctx.AddIntermediateVariable(NumberDataViewType.Int64, "LabelEncoderOutput", true);
 
-                if (info.TypeSrc.GetItemType().Equals(TextDataViewType.Instance))
+                var type = info.TypeSrc.GetItemType();
+                if (type.Equals(TextDataViewType.Instance))
                 {
                     node = ctx.CreateNode(opType, srcVariableName, labelEncoderOutput, ctx.GetNodeName(opType));
                     var terms = GetTermsAndIds<ReadOnlyMemory<char>>(iinfo, out termIds);
                     node.AddAttribute("keys_strings", terms);
                 }
-                else if (info.TypeSrc.GetItemType().Equals(NumberDataViewType.Single))
+                else if (type.Equals(NumberDataViewType.Single))
                 {
                     node = ctx.CreateNode(opType, srcVariableName, labelEncoderOutput, ctx.GetNodeName(opType));
                     var terms = GetTermsAndIds<float>(iinfo, out termIds);
                     node.AddAttribute("keys_floats", terms);
                 }
-                else if (info.TypeSrc.GetItemType().Equals(NumberDataViewType.Double))
+                else if (type.Equals(NumberDataViewType.Double))
                 {
                     // LabelEncoder doesn't support double tensors, so values are cast to floats
                     var castOutput = ctx.AddIntermediateVariable(NumberDataViewType.Single, "castOutput", true);
-                    castNode = ctx.CreateNode("Cast", srcVariableName, castOutput, ctx.GetNodeName(opType), "");
+                    castNode = ctx.CreateNode("Cast", srcVariableName, castOutput, ctx.GetNodeName("Cast"), "");
                     var t = InternalDataKindExtensions.ToInternalDataKind(DataKind.Single).ToType();
                     castNode.AddAttribute("to", t);
                     node = ctx.CreateNode(opType, castOutput, labelEncoderOutput, ctx.GetNodeName(opType));
                     var terms = GetTermsAndIds<double>(iinfo, out termIds);
                     node.AddAttribute("keys_floats", terms);
                 }
-                else if (info.TypeSrc.GetItemType().Equals(NumberDataViewType.Int64))
+                else if (type.Equals(NumberDataViewType.Int64))
                 {
                     // LabelEncoder doesn't support mapping int64 -> int64, so values are cast to strings
                     var castOutput = ctx.AddIntermediateVariable(TextDataViewType.Instance, "castOutput", true);
-                    castNode = ctx.CreateNode("Cast", srcVariableName, castOutput, ctx.GetNodeName(opType), "");
+                    castNode = ctx.CreateNode("Cast", srcVariableName, castOutput, ctx.GetNodeName("Cast"), "");
```

Comment:
Are the first five lines of each of these blocks the same? If so, can they be refactored better?
Even better, can you refactor all these into a single function that is a wrapper with a generic type around GetTermsAndIds?